### ISSUE TITLE
Failed to invoke procedure `gds.graph.create.cypher` ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by nerrve** [neo4j/graph-data-science#77]
I received the following exception while creating a graph of 2 nodes.
The following is the cypher i fired

```
CALL gds.graph.create.cypher(
'cluster111',
'match(c:label1)-[:rel1]->(x) where id(c) = 14194 return distinct id(x) as id',
'match(c:label1)-[:rel1]->(x)-[:rel2]->(y) where (y)<-[:rel1]-(c) and id(c) = 14194 return id(x) as source, id(y) as target',
{
}
) YIELD nodeCount AS nc
```

The graph is like this
(label1)-[:rel1]->(x) 
(2 such nodes of type x and 1 node of type label1)

The following is the stacktrace
```
java.lang.ArrayIndexOutOfBoundsException: Index 10 out of bounds for length 10
        at java.base/java.lang.invoke.VarHandle$1.apply(VarHandle.java:2011)
        at java.base/java.lang.invoke.VarHandle$1.apply(VarHandle.java:2008)
        at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:159)
        at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:156)
        at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:62)
        at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
        at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
        at java.base/java.lang.invoke.VarHandleObjects$Array.getVolatile(VarHandleObjects.java:438)
        at java.base/java.lang.invoke.VarHandleGuards.guard_LI_L(VarHandleGuards.java:646)
        at java.base/java.util.concurrent.atomic.AtomicReferenceArray.get(AtomicReferenceArray.java:100)
        at org.neo4j.graphalgo.core.utils.paged.HugeSparseLongArray$Builder.setIfAbsent(HugeSparseLongArray.java:153)
        at org.neo4j.graphalgo.core.loading.IdMapBuilder.addAndCheckNodes(IdMapBuilder.java:131)
        at org.neo4j.graphalgo.core.loading.IdMapBuilder.lambda$addChecked$2(IdMapBuilder.java:96)
        at org.neo4j.graphalgo.core.concurrency.ParallelUtil.readParallel(ParallelUtil.java:248)
        at org.neo4j.graphalgo.core.loading.IdMapBuilder.buildSparseNodeMapping(IdMapBuilder.java:87)
        at org.neo4j.graphalgo.core.loading.IdMapBuilder.buildChecked(IdMapBuilder.java:65)
        at org.neo4j.graphalgo.core.loading.CypherNodeLoader.result(CypherNodeLoader.java:120)
        at org.neo4j.graphalgo.core.loading.CypherNodeLoader.result(CypherNodeLoader.java:48)
        at org.neo4j.graphalgo.core.loading.CypherRecordLoader.load(CypherRecordLoader.java:86)
        at org.neo4j.graphalgo.core.loading.CypherFactory.lambda$build$1(CypherFactory.java:127)
        at org.neo4j.graphalgo.core.SecureTransaction.apply(SecureTransaction.java:131)
        at org.neo4j.graphalgo.core.loading.CypherFactory.build(CypherFactory.java:111)
        at org.neo4j.graphalgo.core.GraphLoader.graphStore(GraphLoader.java:47)
        at org.neo4j.graphalgo.catalog.GraphCreateProc.createGraph(GraphCreateProc.java:164)
        at org.neo4j.graphalgo.catalog.GraphCreateProc.lambda$create$1(GraphCreateProc.java:129)
        at org.neo4j.graphalgo.BaseProc.runWithExceptionLogging(BaseProc.java:98)
        at org.neo4j.graphalgo.catalog.GraphCreateProc.create(GraphCreateProc.java:127)
        at org.neo4j.kernel.impl.proc.GeneratedProcedure_cypher66263696650598.apply(Unknown Source)
        at org.neo4j.procedure.impl.ProcedureRegistry.callProcedure(ProcedureRegistry.java:208)
        at org.neo4j.procedure.impl.GlobalProceduresRegistry.callProcedure(GlobalProceduresRegistry.java:323)
        at org.neo4j.kernel.impl.newapi.AllStoreHolder.callProcedure(AllStoreHolder.java:966)
        at org.neo4j.kernel.impl.newapi.AllStoreHolder.procedureCallRead(AllStoreHolder.java:869)
        at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$.$anonfun$callReadOnlyProcedure$2(CallSupport.scala:51)
        at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$.callProcedure(CallSupport.scala:98)
        at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$.callReadOnlyProcedure(CallSupport.scala:53)
        at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext.callReadOnlyProcedure(TransactionBoundQueryContext.scala:880)
        at org.neo4j.cypher.internal.planning.ExceptionTranslatingQueryContext.callReadOnlyProcedure(ExceptionTranslatingQueryContext.scala:203)
        at org.neo4j.cypher.internal.runtime.LazyReadOnlyCallMode.callProcedure(ProcedureCallMode.scala:52)
        at org.neo4j.cypher.internal.runtime.interpreted.pipes.ProcedureCallPipe.call(ProcedureCallPipe.scala:87)
        at org.neo4j.cypher.internal.runtime.interpreted.pipes.ProcedureCallPipe.$anonfun$internalCreateResultsByAppending$1(ProcedureCallPipe.scala:73)
        at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
        at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:458)
        at org.neo4j.cypher.internal.runtime.interpreted.PipeExecutionResult.serveResults(PipeExecutionResult.scala:74)
        at org.neo4j.cypher.internal.runtime.interpreted.PipeExecutionResult.request(PipeExecutionResult.scala:62)
        at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.request(StandardInternalExecutionResult.scala:99)
```

**Comment by s1ck** [comment]
@nerrve Thanks for reporting the issue. It seems you're running on a GDS version `<= 1.3.x`. Could you please share the exact version you're running on? We recently made some changes in the above code path. I tried to reproduce your problem with the following small graph:
```
CREATE
  (a:label1 {id: 0})
, (x1)
, (x2)
, (y)
, (a)-[:rel1]->(x1)
, (a)-[:rel1]->(x2)
, (x2)-[:rel1]->(y)
, (a)-[:label1]->(y);
```
node-query `match (c:label1)-[:rel1]->(x) where c.id = 0 return distinct id(x) as id`
rel-query `match (c:label1)-[:rel1]->(x)-[:rel2]->(y) where (y)<-[:label1]-(c) and c.id = 0 return id(x) as source, id(y) as target`

Would be great if you could also share your example graph, that would help the investigation. Thanks.

**Comment by nerrve** [comment]
I'm so sorry, i forgot to mention the version.
I'm using neo4j-graph-data-science-1.3.1-standalone.jar on neo4j-4.1.1

I've modified the sample graph below

```
CREATE
  (a:cluster {id: 0})
, (x1:entity)
, (x2:entity)
, (a)-[:has]->(x1)
, (a)-[:has]->(x2)
, (x1)-[:relatesTo]->(x2);
```

**Comment by s1ck** [comment]
I tried to reproduce the issue with no success. I also noticed that in your example graph, there is no relationship type called `label1`. However, in your relationship query, you add a constraint `where (y)<-[:label1]-(c)`. Even if you replace `label1` with `rel2` (I guess that's what you meant), you wouldn't get any relationship because the `c` node in your pattern only matches the `a` node in the input graph and that one doesn't have any incoming relationships.
